### DISCUSSION TITLE
FreshRSS as an IndieWeb community platform

### DISCUSSION THREAD

**Discussion by connectedblue** [FreshRSS/FreshRSS#6609] [Show and tell]
First of all, thank you for producing an amazing, feature rich platform.  I use FreshRSS several times a day to look at my 600+ feeds and it behaves flawlessly.

However I am also using this to power a new community platform for IndieWeb artists:

https://artocalypse.org/

It's perfect for encouraging people to post on their own site and we display the aggregated feeds.  We also use the label feature to curate expos.  And #hashtagging creates a magical "cross site" feed that artists can use to create their own collaboration feeds by simply agreeing to tag their own post with an agreed tag.

I have disabled search for the moment, but plan to introduce it again very soon for a specific purpose.

I made a few CSS changes to get the look, but only really a handful of php customisations.  I keep them in a separate directory and recently did the upgrade from 1.23 to 1.24 which only took a few hours to merge in (next time I anticipate it will only be an hour or two since I now have a merge process written down).  So I'm super happy that I can stay current with upstream and apply the small changes for this platform.

At some point in the future, I'd like to open source my customisations to enable others to create similar community public feed platforms.  I think it's a very, very powerful tool for indieweb communities.

**Comment by Alkarex** [comment]
Looks great, and thanks for the feedback!
If needed, we could add some more hooks so that such a system can be implemented purely from a FreshRSS extension instead of having to apply patches. Feel free to express your needs

**Reply by connectedblue** [reply]
Nice ... there are three broad areas which could be generically useful:

- add a public class to non-logged in users to allow the anonymous admin feed to be styled differently in a theme.  This is a trivial patch which I could submit as a PR if you wish.
- add a hook to put custom content in the category/label div in place of the name
- add a custom header above the FreshRSS one

I don't really know the codebase well enough to know how to write extensions, so can't really propose a solution.

Some of the other changes were hacks specific to me (change the order to date on post rather than date seen, remove the unread number from the favicon and add in a "close" bar when articles are unfurled.
